webui/react/src/components/kit/useTags.tsx

```diff
@@ -0,0 +1,9 @@
+import React, { useState } from 'react';
+
+import Tags, { Props, tagsActionHelper } from 'components/kit/Tags';
+
+export const useTags = (tags: string[]): ((props?: Omit<Props, 'tags'>) => JSX.Element) => {
```

Comment:
I would like to keep this hook because it acts like a wrapper to reduce code duplication. When we actually use the `<Tags>` through out the app we use the controlled version, but in UI kit, we'd like to use the uncontrolled version, so basically the hook controls the state and the component (in this case the `TagsSection` in UI kit) no longer needs to.

Reply:
I think I prefer consistency because it would be ideal if devs could also use the DesignKit page as a guide for how to write components in the app. Just want to avoid a situation where we have to manage usage of both versions throughout the app because some devs use the hook and some use the component directly. also kind of confusing that `useExperimentTags` returns a function helper and `useTags` returns a component. None of this is a huge problem, but I'm not sure it's worth it just to avoid code duplication in one place -- seems like not a big deal to use state and import the helper in `TagsSection`.

just suggestions though, I don't think this is worth blocking a merge so ultimately it's up to you.

Reply:
FWIW something like `useTags` is pretty standard. We're essentially providing both a controlled and uncontrolled version of the component here which helps consistency by creating a component that supports both complicated and simple usecases.